MapboxDirections/MBDirectionsRouter.swift

```diff
@@ -48,83 +49,65 @@ internal enum MBDirectionsRouter: Router {
         case Full = "full"
     }
     
-    case V4(Configuration, String, [MBDirectionsWaypoint], Bool?, InstructionFormat?, GeometryFormatV4?, Bool?)
-    case V5(Configuration, String, [MBDirectionsWaypoint], Bool?, GeometryFormatV5?, OverviewGranularity?, Bool?, Bool?)
-    
-    var method: HTTPMethod {
-        return .GET
-    }
-    
-    var encoding: HTTPEncoding {
-        return .URL
-    }
-    
-    var configuration: Configuration {
-        switch self {
-        case .V4(let config, _, _, _, _, _, _): return config
-        case .V5(let config, _, _, _, _, _, _, _): return config
-        }
-    }
+    case V4(MBDirectionsConfiguration, String, [MBDirectionsWaypoint], Bool?, InstructionFormat?, GeometryFormatV4?, Bool?)
+    case V5(MBDirectionsConfiguration, String, [MBDirectionsWaypoint], Bool?, GeometryFormatV5?, OverviewGranularity?, Bool?, Bool?)
     
-    var params: [String: String] {
-        switch self {
-        case .V4(_, _, _, let includeAlternatives, let instructionFormat, let geometryFormat, let includeSteps):
-            var params: [String: String] = [
-                "alternatives": String(includeAlternatives ?? false),
-            ]
-            if let instructionFormat = instructionFormat {
-                params["instructions"] = instructionFormat.rawValue
-            }
-            if let geometryFormat = geometryFormat {
-                params["geometry"] = geometryFormat.rawValue
-            }
-            if let includeSteps = includeSteps {
-                params["steps"] = String(includeSteps)
-            }
-            return params
-            
-        case .V5(_, _, let waypoints, let includeAlternative, let geometryFormat, let overviewGranularity, let includeSteps, let allowPointUTurns):
-            var params: [String: String] = [:]
-            if let includeAlternative = includeAlternative {
-                params["alternative"] = String(includeAlternative)
-            }
-            let hasHeadings = !(waypoints.flatMap { $0.heading }.isEmpty)
-            if hasHeadings {
-                params["bearings"] = waypoints.map {
-                    return $0.heading != nil ? "\($0.heading!)" : ""
-                }.joinWithSeparator(";")
+    var URLRequest: NSMutableURLRequest {
+        let result: (configuration: MBDirectionsConfiguration, path: String, parameters: [String: AnyObject]) = {
+            switch self {
```

Comment:
Instead of this massive assignment block, declare `result` with `let` but don't assign anything to it on that line. Then write to the variable in every case of a switch statement before ever reading from the variable.